### PR TITLE
[4] Folder creation - Feedback INTL and ICONV PHP Extensions not installed

### PR DESCRIPTION
**Pull request by PhilETaylor**
Pull Request for Issue #33449 .

### Summary of Changes

If you are on a server where ICONV AND INTL extensions are NOT installed and you try to create a folder in com_media with name `ыыыыыыыы` then it will fail, but tell you that it created the folder. 

This PR handles that, and tells you the real reason. 

### Testing Instructions

On a server where ICONV AND INTL extensions are NOT installed and you try to create a folder in com_media with name `ыыыыыыыы` then it will fail, but tell you that it created the folder. 

Apply PR

On a server where ICONV AND INTL extensions are NOT installed and you try to create a folder in com_media with name `ыыыыыыыы` then it will fail, but tell you that it failed to create the folder. 

### Actual result BEFORE applying this Pull Request

Screenshot 2021-04-30 at 22 22 46

### Expected result AFTER applying this Pull Request

Screenshot 2021-04-30 at 22 22 26

### Documentation Changes Required

none. 

Isolated code test proving this issue available here https://3v4l.org/BRcNa

### CODE REVIEW

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 1fd487c7944ba84ecc022c5fc893a57f2135853f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33451">issues.joomla.org/tracker/joomla-cms/33451</a>.</sub>

**Comment by Kostelano** [comment]
Do we need the same language constant in the frontend?

**Comment by PhilETaylor** [comment]
probably. Im only doing half jobs tonight :)

**Comment by PhilETaylor** [comment]
Frontend seems to use the backend lang constants. 

Screenshot 2021-04-30 at 23 06 05

File: administrator/language/en-GB/com_media.ini [hunk 1]
```diff
@@ -17,6 +17,7 @@ COM_MEDIA_CONFIRM_DELETE_MODAL="Delete"
 COM_MEDIA_CONFIRM_DELETE_MODAL_HEADING="Confirm Delete"
 COM_MEDIA_CREATE_NEW_FOLDER="Create New Folder"
 COM_MEDIA_CREATE_NEW_FOLDER_ERROR="Error creating folder."
+COM_MEDIA_CREATE_NEW_FOLDER_ERROR_MAKESAFE="Could not make the folder name safe. If it contains non ASCII chars please ensure you have the INTL and ICONV PHP Extensions installed."
```
Comment (Quy) [inline administrator/language/en-GB/com_media.ini]:
```suggestion
COM_MEDIA_CREATE_NEW_FOLDER_ERROR_MAKESAFE="Could not make the folder name safe. If it contains non ASCII characters, please ensure that the INTL and ICONV PHP Extensions are installed."
```

**Comment by richard67** [comment]
I've restored @Kostelano 's test result in the issue tracker so it's properly counted again, since the commit which invalidated the count was just a change in the language string for better English.

**Comment by richard67** [comment]
@PhilETaylor There is already a language string with (almost) the same text as the one added by you in this PR here. It has added by you with your PR #33454 here:
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/language/en-GB/com_media.ini#L72

Beside that it has been added ad the wrong place regarding alphabetical order, we should use one common string for the same message from the other PR and this one here.

I leave it up to you which one you use, but I'd prefer to use the one added by this PR here, remove the other one and change to the new sting at the following 2 places:
https://github.com/PhilETaylor/joomla-cms/blob/4.0-dev/plugins/filesystem/local/src/Adapter/LocalAdapter.php#L552

https://github.com/PhilETaylor/joomla-cms/blob/4.0-dev/plugins/filesystem/local/src/Adapter/LocalAdapter.php#L783

**Comment by richard67** [comment]
Another problem I have: Since PR #33454 has been merged, I can't reproduce issue #33449 anymore. When attempting to create a new folder with only cyrillic characters and intl not available, I get the new message introduced with PR #33454 , even if that PR says it was for copying only:



But from code review I'd say this PR here is ok for me if the issue with the language strings I've mentioned above is solved.

**Pull request closed without merging**